python/python/glide/async_commands/server_modules/ft_options/ft_aggregate_options.py

```diff
@@ -0,0 +1,322 @@
+            properties (List[TEncodable]): The list of properties to be used for grouping the results in the pipeline.
+            reducers (Optional[List[Reducer]]): The list of functions that handles the group entries by performing multiple aggregate operations.
+        """
+        self.properties = properties
+        self.reducers = reducers
+
+    def toArgs(self) -> List[TEncodable]:
+        """
+        Get the arguments for the GroupBy clause.
+
+        Returns:
+            List[TEncodable]: A list arguments for GroupBy clause.
+        """
+
+        args = super().toArgs()
+        args.append(FtAggregateKeywords.GROUPBY)
+        if self.properties:
+            args.append(str(len(self.properties)))
+            args.extend(self.properties)
+        if self.reducers:
+            for reducer in self.reducers:
+                args.extend(reducer.toArgs())
+        return args
+
+
+class SortOrder(Enum):
+    """
+    All possible values for the sort order for the SortBy clause.
+    """
+
+    ASC = "ASC"
+    """
+    For sorting the results in ascending order.
+    """
+    DESC = "DESC"
+    """
+    For sorting the results in descending order.
+    """
+
+
+class SortByProperty:
+    """
+    This class represents the a single property for the SortBy clause.
+    """
+
+    def __init__(self, property: TEncodable, order: SortOrder):
+        """
+        Initialize a new SortByProperty instance.
+
+        Args:
+            property (TEncodable): The sorting parameter.
+            order (SortOrder): The order for the sorting. This option can be added for each property.
+        """
+        self.property = property
+        self.order = order
+
+    def toArgs(self) -> List[TEncodable]:
+        """
+        Get the arguments for the SortBy clause property.
+
+        Returns:
+            List[TEncodable]: A list of arguments for the SortBy clause property.
+        """
+        args: List[TEncodable] = []
+        args.append(self.property)
+        args.append(self.order.value)
+        return args
+
+
+class SortBy(FtAggregateClause):
+    """
+    A clause for sorting the pipeline up until the point of SORTBY, using a list of properties.
+    """
+
+    def __init__(self, properties: List[SortByProperty], max: Optional[int] = None):
+        """
+        Initialize a new SortBy instance.
+
+        Args:
+            properties (List[SortByProperty]): A list of sorting parameters for the sort operation.
+            max: (Optional[int]): The MAX value for optimizing the sorting, by sorting only for the n-largest elements.
+        """
+        self.properties = properties
+        self.max = max
+
+    def toArgs(self) -> List[TEncodable]:
+        """
+        Get the arguments for the SortBy clause.
+
+        Returns:
+            List[TEncodable]: A list of arguments for the SortBy clause.
+        """
+        args = super().toArgs()
+        args.append(FtAggregateKeywords.SORTBY)
+        if self.properties:
```

Comment:
this condition is always true - please optimize

Reply:
Updated.